src/OpenTelemetry/Metrics/MeterProviderBuilderExtensions.cs

```diff
@@ -121,6 +121,53 @@ public static MeterProviderBuilder AddView(this MeterProviderBuilder meterProvid
             return meterProviderBuilder;
         }
 
+        /// <summary>
+        /// Sets the maximum number of Metric streams supported by the MeterProvider.
+        /// When no Views are configured, every instrument will result in one metric stream,
+        /// so this control the numbers of instruments supported.
+        /// When Views are configued, a single instrument can result in multiple metric stream,
+        /// so this controls the number of streams.
+        /// </summary>
+        /// <param name="meterProviderBuilder">MeterProviderBuilder instance.</param>
+        /// <param name="metricStreamLimit">The number of metric streams allowed.</param>
+        /// <returns>Returns <see cref="MeterProviderBuilder"/> for chaining.</returns>
+        /// <remarks>
+        /// If an instrument is created, but disposed later, this will still be contributing to the limit.
+        /// This may change in the future.
+        /// </remarks>
+        public static MeterProviderBuilder SetMetricStreamLimit(this MeterProviderBuilder meterProviderBuilder, int metricStreamLimit)
+        {
+            if (meterProviderBuilder is MeterProviderBuilderBase meterProviderBuilderBase)
+            {
+                meterProviderBuilderBase.SetMetricStreamLimit(metricStreamLimit);
+            }
+
+            return meterProviderBuilder;
+        }
+
+        /// <summary>
+        /// Sets the maximum number of MetricPoints allowed per metric stream.
+        /// This limis the number of unique combination of key value pairs used
+        /// for reporting measurement.
+        /// </summary>
+        /// <param name="meterProviderBuilder">MeterProviderBuilder instance.</param>
+        /// <param name="metricPointLimit">The maximum number of metric points allowed per metric stream.</param>
+        /// <returns>Returns <see cref="MeterProviderBuilder"/> for chaining.</returns>
+        /// <remarks>
+        /// If a particular combination of key value pair is used atleast once,
+        /// it will still be contributing to the limit.
+        /// This may change in the future.
```

Comment:
Not sure I understand this remark.

Reply:
Current behaviour is - if you ever use a KVP, then its counted against the cap. Once https://github.com/open-telemetry/opentelemetry-dotnet/issues/2360 is addressed, we'll be able to totally forgot unused ones, and then it'd no longer contribute to the cap.

Reply:
Gotcha, figured this is what you meant. What do you think about this clarification https://github.com/open-telemetry/opentelemetry-dotnet/pull/2635#discussion_r752541574